industrial_robot_client/src/joint_trajectory_action.cpp

```diff
@@ -123,6 +191,24 @@ void JointTrajectoryAction::goalCB(JointTractoryActionServer::GoalHandle gh)
     return;
   }
 
+  // check robot can actually execute trajectory, if not, refuse goal.
+  // no point in accepting the goal only to cancel it immediately later
+  if (!is_motion_server_ok(last_robot_status_, consider_status_unknowns_ok_) && !ignore_motion_server_error_)
```

Comment:
micro optimization ahead: check the bool first?

Reply:
I think I'll leave it like this. It's indeed a very minor optimisation, and the current order reads almost like plain English: "if the motion server is not ok AND we don't ignore motion server errors, ...".